snap/validate.go

```diff
@@ -403,10 +403,16 @@ func validateAppOrderCycles(apps map[string]*AppInfo) error {
 
 	for _, app := range apps {
 		for _, other := range app.After {
+			if strings.HasPrefix(other, "external:") {
+				continue
+			}
 			predecessors[app.Name]++
 			successors[other] = append(successors[other], app.Name)
 		}
 		for _, other := range app.Before {
+			if strings.HasPrefix(other, "external:") {
```

Comment:
Wouldn't skipping the check allow this to go unnoticed?
```
app-1:
    before: [external:snap.seeded]
    after: [app-2]
app-2:
    after: [external:snap.seeded]
```
I think we could leave it as a TODO, but the phase when successors/predecessors and the queues are built, would need to account for those external apps, i.e. instead of ranging over `apps`, range over `append(apps, tempApps)` where `tempApps` are created ad hoc from `external:` services